samples/nrf9160/nrf_cloud_coap_client/prj.conf

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+# General config
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+
+CONFIG_DK_LIBRARY=y
+
+# Mbed TLS logs include %zu formatters, which are not handled by the nano
+# version of newlib
+CONFIG_NEWLIB_LIBC_NANO=n
+
+# Main thread
+CONFIG_MAIN_STACK_SIZE=6144
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=6144
+
+# Networking
+CONFIG_NETWORKING=y
+CONFIG_NET_NATIVE=n
+CONFIG_NET_OFFLOAD=y
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_OFFLOAD=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+
+# LTE link control
+CONFIG_LTE_LINK_CONTROL=y
+CONFIG_LTE_AUTO_INIT_AND_CONNECT=n
+CONFIG_LTE_NETWORK_MODE_LTE_M_GPS=y
+
+# Modem library
+CONFIG_NRF_MODEM_LIB=y
+CONFIG_NRF_MODEM_LIB_LOG_FW_VERSION_UUID=n
+CONFIG_MODEM_JWT=y
+
+# Modem info
+CONFIG_MODEM_INFO=y
+CONFIG_MODEM_INFO_ADD_DEVICE=y
+CONFIG_MODEM_INFO_ADD_NETWORK=y
+CONFIG_MODEM_INFO_ADD_DATE_TIME=y
+CONFIG_MODEM_INFO_ADD_SIM=y
+CONFIG_MODEM_INFO_ADD_SIM_ICCID=y
+CONFIG_MODEM_INFO_ADD_SIM_IMSI=n
+
+# AT Host
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_0_INTERRUPT_DRIVEN=y
+CONFIG_AT_HOST_LIBRARY=y
+CONFIG_AT_MONITOR_HEAP_SIZE=4096
+
+# LTE link control
+CONFIG_NET_L2_VIRTUAL=y
+
+# Date/Time
+CONFIG_DATE_TIME=y
+CONFIG_DATE_TIME_AUTO_UPDATE=y
+CONFIG_DATE_TIME_MODEM=y
+CONFIG_DATE_TIME_NTP=n
+
+CONFIG_FPU=y
+CONFIG_CBPRINTF_FP_SUPPORT=y
+
+CONFIG_COAP_CLIENT=y
+CONFIG_COAP_CLIENT_BLOCK_SIZE=1024
+CONFIG_COAP_CLIENT_STACK_SIZE=6144
+CONFIG_COAP_CLIENT_THREAD_PRIORITY=0
+CONFIG_COAP_EXTENDED_OPTIONS_LEN=y
+CONFIG_COAP_EXTENDED_OPTIONS_LEN_VALUE=22
+
+CONFIG_DOWNLOAD_CLIENT_STACK_SIZE=4096
+
+CONFIG_NRF_CLOUD_COAP=y
+CONFIG_NRF_CLOUD_REST=y
```

Comment:
Does REST need to be enabled to use CoAP and why?

Reply:
I was using 2 REST functions and some structures to save time getting you something to work with.

I eliminated one REST function today and will see about the other.  But regardless, it builds without CONFIG_NRF_CLOUD_REST, so I can remove that from prj.conf.